pycbc/io/hdf.py

```diff
@@ -1028,41 +1028,38 @@ def to_coinc_hdf_object(self, file_name):
         # Some fields are special cases
         logging.info("Outputting search results")
         time = self.get_end_time()
-        ofd.create_dataset('time', data=time, dtype=np.float32)
+        ofd['time'] = time
 
         if self._inclusive:
             ifar = self.get_coincfile_array('ifar')
-            ofd.create_dataset('ifar', data=ifar, dtype=np.float32)
+            ofd['ifar'] = ifar
 
         ifar_exc = self.get_coincfile_array('ifar_exc')
-        ofd.create_dataset('ifar_exclusive', data=ifar_exc,
-                           dtype=np.float32)
+        ofd['ifar_exclusive'] = ifar_exc
 
         if self._inclusive:
             fap = self.get_coincfile_array('fap')
-            ofd.create_dataset('p_value', data=fap,
-                               dtype=np.float32)
+            ofd['p_value'] = fap
 
         fap_exc = self.get_coincfile_array('fap_exc')
-        ofd.create_dataset('p_value_exclusive', data=fap_exc,
-                           dtype=np.float32)
+        ofd['p_value_exclusive'] = fap_exc
 
         # Coinc fields
         for field in ['stat']:
             vals = self.get_coincfile_array(field)
-            ofd.create_dataset(field, data=vals, dtype=np.float32)
+            ofd[field] = vals
```

Comment:
I don't think we need to define 'vals' any more, just assign the dataset directly